tests/templates/test_subroutines/test_kupccgsd.py

```diff
@@ -624,3 +624,20 @@ def test_torch(self, tol):
         grads2 = [weights.grad]
 
         assert np.allclose(grads[0], grads2[0], atol=tol, rtol=0)
+
+
+class TestGradient:
```

Comment:
Yep! Perfect

Reply:
Although it looks like the test is failing, because this test file doesn't import PennyLane NumPy:

```python
  File "/home/runner/work/pennylane/pennylane/tests/templates/test_subroutines/test_kupccgsd.py", line 640, in test_ps_rule_gradient
    weights = np.array([0.55, 0.72, 0.6, 0.54, 0.42, 0.65], requires_grad=True).reshape(1, -1)
TypeError: array() got an unexpected keyword argument 'requires_grad'
```

Insteead you could do `qml.numpy.array(...)` inside this function, and that should work.

Reply:
Hmm... still failing

https://github.com/PennyLaneAI/pennylane/runs/7748511970?check_suite_focus=true#step:11:467